File: WinForms/CS/TreeMapMainDemo/Modules/Sunburst/SunburstHierarchicalData.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Windows.Forms;
using System.Xml.Linq;
using DevExpress.TreeMap;
using DevExpress.Utils;

namespace DevExpress.XtraTreeMap.Demos {
    public partial class SunburstHierarchicalData : TreeMapDemoModuleWithOptions {
        SunburstHierarchicalDataAdapter HierarchicalDataAdapter { get { return (SunburstHierarchicalDataAdapter)sunburstControl1.DataAdapter; } }

        public override Control ActiveDemoModule { get { return sunburstControl1; } }

        public SunburstHierarchicalData() {
            InitializeComponent();
            InitializeControls(sunburstControl1, colorizerCBE, varyColorInGroupLCI, modeLCI);
            HierarchicalDataAdapter.DataSource = LoadDataFromXML();
            HierarchicalDataAdapter.Mappings[0].Type = typeof(TypeInfo);
        }
        void ToolTipController_BeforeShow(object sender, ToolTipControllerShowEventArgs e) {
            ISunburstItem sunburstItem = (ISunburstItem)e.SelectedObject;
            string namespaceString = "DevExpress.XtraBars";
            foreach(ISunburstItem item in sunburstControl1.GetItemPath(sunburstItem))
                namespaceString += "." + item.Label.ToString();
            SuperToolTip superToolTip = new SuperToolTip() { MaxWidth = 400 };
            superToolTip.Items.Add(new ToolTipTitleItem() { Text = namespaceString });
            superToolTip.Items.Add(new ToolTipSeparatorItem());
            superToolTip.Items.Add(new ToolTipItem() { Text = "Types Count: " + sunburstItem.Value });

            e.SuperTip = superToolTip;
        }
        void LoadData(XElement element, List<TypeInfo> datas) {
            TypeInfo data = new TypeInfo() { NamespaceString = element.Element("Namespace").Value, TypesCount = Convert.ToDouble(element.Element("TypesCount").Value, CultureInfo.InvariantCulture) };
            datas.Add(data);
            foreach(XElement item in element.Element("NestedNamespaces").Elements())
                LoadData(item, data.NestedNamespaces);
        }
        List<TypeInfo> LoadDataFromXML() {
            XDocument document = XDocument.Load(Utils.GetRelativePath("XtraBarsTypes.xml"));
            List<TypeInfo> datas = new List<TypeInfo>();
            if(document != null) {
                foreach(XElement element in document.Element("ArrayOfTypeInfo").Elements())
                    LoadData(element, datas);
            }
            return datas;
        }
    }

    public class TypeInfo {
        readonly List<TypeInfo> _nestedNamespaces = new List<TypeInfo>();

        public string NamespaceString { get; set; }
        public double TypesCount { get; set; }
        public List<TypeInfo> NestedNamespaces { get { return _nestedNamespaces; } }
    }
}

File: WinForms/VB/TreeMapMainDemo/Modules/Sunburst/SunburstHierarchicalData.vb
Imports System
Imports System.Collections.Generic
Imports System.Globalization
Imports System.Windows.Forms
Imports System.Xml.Linq
Imports DevExpress.TreeMap
Imports DevExpress.Utils

Namespace DevExpress.XtraTreeMap.Demos

    Public Partial Class SunburstHierarchicalData
        Inherits TreeMapDemoModuleWithOptions

        Private ReadOnly Property HierarchicalDataAdapter As SunburstHierarchicalDataAdapter
            Get
                Return CType(sunburstControl1.DataAdapter, SunburstHierarchicalDataAdapter)
            End Get
        End Property

        Public Overrides ReadOnly Property ActiveDemoModule As Control
            Get
                Return sunburstControl1
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
            InitializeControls(sunburstControl1, colorizerCBE, varyColorInGroupLCI, modeLCI)
            HierarchicalDataAdapter.DataSource = LoadDataFromXML()
            HierarchicalDataAdapter.Mappings(0).Type = GetType(TypeInfo)
        End Sub

        Private Sub ToolTipController_BeforeShow(ByVal sender As Object, ByVal e As ToolTipControllerShowEventArgs)
            Dim sunburstItem As ISunburstItem = CType(e.SelectedObject, ISunburstItem)
            Dim namespaceString As String = "DevExpress.XtraBars"
            For Each item As ISunburstItem In sunburstControl1.GetItemPath(sunburstItem)
                namespaceString += "." & item.Label.ToString()
            Next

            Dim superToolTip As SuperToolTip = New SuperToolTip() With {.MaxWidth = 400}
            superToolTip.Items.Add(New ToolTipTitleItem() With {.Text = namespaceString})
            superToolTip.Items.Add(New ToolTipSeparatorItem())
            superToolTip.Items.Add(New ToolTipItem() With {.Text = "Types Count: " & sunburstItem.Value})
            e.SuperTip = superToolTip
        End Sub

        Private Sub LoadData(ByVal element As XElement, ByVal datas As List(Of TypeInfo))
            Dim data As TypeInfo = New TypeInfo() With {.NamespaceString = element.Element("Namespace").Value, .TypesCount = Convert.ToDouble(element.Element("TypesCount").Value, CultureInfo.InvariantCulture)}
            datas.Add(data)
            For Each item As XElement In element.Element("NestedNamespaces").Elements()
                LoadData(item, data.NestedNamespaces)
            Next
        End Sub

        Private Function LoadDataFromXML() As List(Of TypeInfo)
            Dim document As XDocument = XDocument.Load(GetRelativePath("XtraBarsTypes.xml"))
            Dim datas As List(Of TypeInfo) = New List(Of TypeInfo)()
            If document IsNot Nothing Then
                For Each element As XElement In document.Element("ArrayOfTypeInfo").Elements()
                    LoadData(element, datas)
                Next
            End If

            Return datas
        End Function
    End Class

    Public Class TypeInfo

        Private ReadOnly _nestedNamespaces As List(Of TypeInfo) = New List(Of TypeInfo)()

        Public Property NamespaceString As String

        Public Property TypesCount As Double

        Public ReadOnly Property NestedNamespaces As List(Of TypeInfo)
            Get
                Return _nestedNamespaces
            End Get
        End Property
    End Class
End Namespace
